controllers/schedule/pause/controller.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pause
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	k8sError "k8s.io/apimachinery/pkg/api/errors"
+	k8sTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/schedule/utils"
+	"github.com/chaos-mesh/chaos-mesh/controllers/types"
+)
+
+type Reconciler struct {
+	client.Client
+	Log          logr.Logger
+	ActiveLister *utils.ActiveLister
+
+	Recorder record.EventRecorder
+}
+
+func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	schedule := &v1alpha1.Schedule{}
+	err := r.Get(ctx, req.NamespacedName, schedule)
+	if err != nil {
+		if !k8sError.IsNotFound(err) {
+			r.Log.Error(err, "unable to get chaos")
+		}
+		return ctrl.Result{}, nil
+	}
+
+	list, err := r.ActiveLister.ListActiveJobs(ctx, schedule)
+	if err != nil {
+		r.Recorder.Eventf(schedule, "Warning", "Failed", "Failed to list active jobs: %s", err.Error())
+		return ctrl.Result{}, nil
+	}
+
+	items := reflect.ValueOf(list).Elem().FieldByName("Items")
+	for i := 0; i < items.Len(); i++ {
```

Comment:
Do we need to set pause for finished jobs?

Reply:
It doesn't matter. We don't need to filter out finished jobs manually :smile_cat: